common/spec/dependabot/file_parsers/base/dependency_set_spec.rb

```diff
@@ -234,4 +234,140 @@
       end
     end
   end
+
+  context "when multiple versions of a dependency are added" do
+    let(:foo_v1) do
+      Dependabot::Dependency.new(
+        name: "foo",
+        version: "1.0",
+        requirements: [],
+        package_manager: "dummy"
+      )
+    end
+
+    let(:foo_v1_1) do # rubocop:disable Naming/VariableNumber
+      Dependabot::Dependency.new(
+        name: "foo",
+        version: "1.1",
+        requirements: [{
+          requirement: "^1",
+          file: "Dummyfile",
+          groups: nil,
+          source: nil
+        }],
+        package_manager: "dummy"
+      )
+    end
+
+    let(:foo_v1_1_alt) do
+      Dependabot::Dependency.new(
+        name: "foo",
+        version: "1.1",
+        requirements: [{
+          requirement: "^1",
+          file: "Dummyfile.lock",
+          groups: ["prod"],
+          source: {
+            type: "registry",
+            url: "https://registry.dummy.org"
+          }
+        }],
+        package_manager: "dummy"
+      )
+    end
+
+    let(:foo_sha) do
+      Dependabot::Dependency.new(
+        name: "foo",
+        version: "d5ac0584ee9ae7bd9288220a39780f155b9ad4c8",
+        requirements: [{
+          requirement: "^1",
+          file: "Dummyfile.lock",
+          groups: ["prod"],
+          source: {
+            type: "git",
+            url: "https://github.com/acme-inc/foo",
+            branch: nil,
+            ref: "main"
+          }
+        }],
+        package_manager: "dummy"
+      )
+    end
+
+    it "merges each into a single combined dependency" do
+      dependency_set = described_class.new << foo_v1 << foo_sha << foo_v1_1
+
+      expect(dependency_set.dependency_for_name("foo")).to eq(
+        Dependabot::Dependency.new(
+          name: "foo",
+          version: "1.0",
+          requirements: (
+            foo_v1.requirements +
+            foo_sha.requirements +
+            foo_v1_1.requirements
+          ).uniq,
+          package_manager: "dummy"
+        )
+      )
+    end
+
+    it "returns all versions in the order they were added by default" do
```

Comment:
It does! although that's not exactly what's happening here—in this case the hash has one entry and the array of versions on that entry is returned in insertion order.

Reply:
👍 Yup, this is a misplaced comment! I was wondering about the dependency order since it changed from an array to hash and incorrectly read this test description as checking for that.

No concerns about the dependency order though since the Hash is ordered!